### PR TITLE
components: Add useCx

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
Adds a `useCx` hook which creates a `cx` function that is able to take in `SerializedStyles` returned by the `css` function from `@emotion/react`. It also attaches these to the current EmotionCache meaning that it will work with iframes.

## How has this been tested?
Unit tests and storybook.

## Types of changes
New feature.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [N/A] My code follows the accessibility standards. 
- [N/A] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by ciampo — COMMENTED** [review]
Very neat!

I wonder if we should at least add some JSDocs to show a usage example and explain why this hook exists in the first place.

Also, out of curiosity, would mixing different Emotion's "techniques" (`css`, `styles`, `useCx`) work and access the same cache?

File: packages/components/src/utils/hooks/use-cx.ts
Comment (ciampo) [inline packages/components/src/utils/hooks/use-cx.ts:18]: We may not need to access a private property if https://github.com/emotion-js/emotion/pull/2418 gets merged 🤞

File: packages/components/src/utils/hooks/use-cx.ts
Comment (ciampo) [inline packages/components/src/utils/hooks/use-cx.ts:55]: Is there any chance that this interpolation technique may change in emotion and potentially break this hook?

Comment (sarayourfriend) [reply packages/components/src/utils/hooks/use-cx.ts:55]: I doubt it, but we'd notice it quickly I would think. Testing that iframes are working with Emotion should be something we do every Emotion upgrade.

Comment (ciampo) [reply packages/components/src/utils/hooks/use-cx.ts:55]: > Testing that iframes are working with Emotion

This is actually a good point. Should we also add a unit test and/or storybook story to showcase iframe compatibility?

Comment (sarayourfriend) [reply packages/components/src/utils/hooks/use-cx.ts:55]: Oh yeah, that'd be good to do, I'll add it.

**Review by ciampo — APPROVED** [review]
I'm not sure if we should wait for some feedback from the Emotion's team before merging, but otherwise these changes LGTM

File: packages/components/src/utils/hooks/use-cx.ts
Comment (sarayourfriend) [reply packages/components/src/utils/hooks/use-cx.ts:18]: Yeah. Unfortunately it seems that my PR isn't gaining any traction.

Looking at the React internals, they seem to use this property a lot, so I would be surprised if it went away completely in the future: https://github.com/facebook/react/search?q=._context

**Pull request merged**
